core/message_pool.go

```diff
@@ -38,21 +55,25 @@ type BlockTimer interface {
 type MessagePool struct {
 	lk sync.RWMutex
 
-	timer   BlockTimer
-	pending map[cid.Cid]*timedmessage // all pending messages
+	api           MessagePoolAPI
+	validator     MessagePoolValidator
+	pending       map[cid.Cid]*timedmessage // all pending messages
+	addressNonces map[addressNonce]bool
```

Comment:
Document the intended use of this as a set of addressNonce corresponding to all the pending messages.